bolt/include/bolt/Core/MCPlusBuilder.h

```diff
@@ -622,6 +628,40 @@ class MCPlusBuilder {
     return std::make_pair(getNoRegister(), getNoRegister());
   }
 
+  /// Analyzes if a pointer is checked to be valid by the end of BB.
+  ///
+  /// It is possible for pointer authentication instructions not to terminate
+  /// the program abnormally on authentication failure and return some *invalid
+  /// pointer* instead (like it is done on AArch64 when FEAT_FPAC is not
+  /// implemented). This might be enough to crash on invalid memory access
+  /// when the pointer is later used as the destination of load/store or branch
+  /// instruction. On the other hand, when the pointer is not used right away,
+  /// it may be important for the compiler to check the address explicitly not
+  /// to introduce signing or authentication oracle.
+  ///
+  /// If this function returns a (Reg, Inst) pair, then it is known that in any
+  /// successor of BB either
+  /// * Reg is trusted, provided it was safe-to-dereference before Inst, or
```

Comment:
I think that "trusted" isn't defined before this sentence?
I'm assuming that "trusted" means "has been authenticated and program termination is guaranteed if authentication failed"?
Maybe we should have a specific term for that?
It seems to me that "trusted" might be a too-generic term, and becomes too confusing to use for this.
Off the top of my head, I'm thinking maybe "fully-authenticated" might work? Not sure if @jacobbramley has a better suggestion?

Reply:
It seems that "safe-to-dereference" and "trusted" are quite complex terms which are better to keep private to `PAuthGadgetScanner.cpp`. Moreover, the definitions of these properties can evolve over time.

I updated the descriptions to be more like "if the register was last written to by an authentication instruction, the program should be terminated or that authentication should be known to succeed". If the "input" of the checker sequence was produced in any other way (which is probably meaningless, though), this is out of scope of this function.